budget-binder-multiplatform-app/src/jsMain/kotlin/de/hsfl/budgetBinder/compose/category/CategoryComponent.kt

```diff
@@ -57,28 +57,64 @@ fun CategoryComponent(screenState: MutableState<Screen>) {
 //Should be put in own File
 @Composable
 fun Icon(category: Category){
-    val imagePath =
+    val imageName =
         when (category.image) {
-            Category.Image.DEFAULT -> "cart.png"
-            Category.Image.SHOPPING -> "cart.png"
+            Category.Image.SHOPPINGCART -> "shopping_cart"
+            Category.Image.SHOPPINGBASKET -> "shopping_basket"
+            Category.Image.CHECKMARK -> "done"
```

Comment:
Ich würde das ganze in `commonMain` packen und mit der `expect` Funktion in den jeweiligen packages anpasse. So das am Ende über JVM die Icons mit der Composable Funktion `Icon(Icons...)` und über JS mit der span Funktion als Composable. 

**In commonMain**
```kotlin
expect fun CategoryToIcon(icon: Category.Image)
```

**In JVM**
```
@Composable
actual fun CategoryToIcon(icon: Category.Image) {
    when (icon) {
        Category.Image.DEFAULT -> Icon(imageVector = Icons.Default.ShoppingCart, contentDescription = null)
        Category.Image.SHOPPINGCART -> Icon(imageVector = Icons.Default.ShoppingCart, contentDescription = null)
        Category.Image.SHOPPINGBASKET -> Icon(imageVector = Icons.Default.ShoppingBasket, contentDescription = null)
        Category.Image.CHECKMARK -> Icon(imageVector = Icons.Default.Done, contentDescription = null)
        ...
    }
}
```

**In JS**
```kotlin
@Composable
actual fun CategoryToIcon(icon: Category.Image) {
    Span(
        attrs = {
            classes("material-icons")
        }
    ) {
        Text(
            when (icon) {
                Category.Image.DEFAULT -> "cart.png"
                Category.Image.SHOPPINGCART -> "shopping_cart"
                Category.Image.SHOPPINGBASKET -> "shopping_basket"
                Category.Image.CHECKMARK -> "done"
                Category.Image.WRONG -> "dangerous"
                ...
            }
        )
    }
}
```

Somit können wir dann am Ende die Composefunktion easy peasy benutzen, um die Icons von Kategorien zu laden. Und es ist vielleicht bisschen einheitlicher
```kotlin
CategoryToIcon(category.image) // Zeigt das korrekte der Kategorie
```

Habe das ganze getestet und diesen Kommentar zum zweiten mal geschrieben, könnte sonst meine Sachen hier rein pushen.

Reply:
Generell könnte man so Compose Funktionen einheitlich in `commonMain` mit `expect/actual` definieren wie z.B. den Router... diese Logik muss dann nur von nicht Composables auch umgesetzte werden, z.B. iOS. Da macht so etwas aber ja auch sinn

Reply:
Ja klingt gut, kannst du gerne hier reinpushen!